src/modules/manage-team/sections/Integrations.tsx

```diff
@@ -1,19 +1,105 @@
 import React from 'react';
 import { Divider, Header, Icon } from 'semantic-ui-react';
+import gql from 'graphql-tag';
+import { Query } from '@apollo/react-components';
+import { History } from 'history';
+import { toast } from 'react-toastify';
+import { Storage } from '../../../support/storage';
+import settings from '../../../config/settings';
 
-export function IntegrationSections() {
-  return (
-    <div>
-      <Header as="h2">
-        <Icon name="settings" />
-        <Header.Content>
-          Integrations
-          <Header.Subheader>Manage integrations</Header.Subheader>
-        </Header.Content>
-      </Header>
-      <Divider />
-    </div>
-  );
+const queryString = require('query-string');
+
+export const GET_TEAM_INTEGRATIONS = gql`
+    query GetTeamIntegrations($id: ID!) {
+        teamById(id: $id) {
+            slackTeamId
+        }
+    }
+`;
+
+interface TeamIntegrationsResponse {
+  teamById: {
+    slackTeamId: string;
+  }
+}
+
+export interface IntegrationsSectionProps {
+  history: History,
 }
 
-export default IntegrationSections;
+
+export default class IntegrationsSection extends React.Component<IntegrationsSectionProps, any> {
+  slackUrl = 'https://slack.com/oauth/v2/authorize?'
+      + `state=${Storage.getItem(settings.TEAM_ID_TOKEN)}&`
+      + `client_id=${settings.SLACK_CLIENT_ID}&`
+      + '&scope=chat:write,commands,incoming-webhook,chat:write.public&user_scope=reactions:read';
```

Comment:
It feels odd to me that the frontend knows this much about the Slack integration. IMO the backend should be in charge of this, as the backend knows which scopes are required for example. Now, if you would update the backend with a feature that requires a new scope, you'd have to update the client as well.

Reply:
I'll look into this because it's indeed a bit hacky. I'm also abusing the state param to indicate what team the workspace should be connected to which is pretty bad.

Reply:
Fixed, it now makes a request to the back-end which generates the oauth url and redirects the user.